### ISSUE TITLE
Crash when clicked on Home Widget

### ISSUE THREAD

**Issue by Surendrajat** [otakuhqz/Omega#35]
## Description

Clicked on `Home Settings` -> `Home Widget` and boom!

## Expected Behavior

The app should not crash and should show relevant settings.

## Actual Behavior

App crashed.

## Steps to Reproduce

1. Long press on Home Screen.
2. Click on `Home Settings`
3. Click on `Home Widget`
4. Watch it crash

## Device and OS

* Device: Oppo K3
* Android version: 10
* Launcher version: v0.6.1 and latest debug from GH actions
* Rom: ColorOS v7.1

## Logcat

```
Build version: 0.6.3 
Build date: 1981-01-01 01:01:02 
Current date: 2021-01-09 18:40:48 
Device: OPPO CPH1955 
 
Stack trace:  
android.view.InflateException: Binary XML file line #17 in com.saggitt.omega:layout/search_container_workspace: Binary XML file line #17 in com.saggitt.omega:layout/search_container_workspace: Error inflating class com.google.android.apps.nexuslauncher.smartspace.SmartspaceView
Caused by: android.view.InflateException: Binary XML file line #17 in com.saggitt.omega:layout/search_container_workspace: Error inflating class com.google.android.apps.nexuslauncher.smartspace.SmartspaceView
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Constructor.newInstance0(Native Method)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
	at android.view.LayoutInflater.createView(LayoutInflater.java:855)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1008)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:963)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:659)
	at com.saggitt.omega.OmegaLayoutInflater.inflate(OmegaLayoutInflater.kt:94)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:534)
	at com.saggitt.omega.views.SmartspacePreview.inflateView(SmartspacePreview.kt:75)
	at com.saggitt.omega.views.SmartspacePreview.inflateCurrentView(SmartspacePreview.kt:71)
	at com.saggitt.omega.views.SmartspacePreview.onValueChanged(SmartspacePreview.kt:63)
	at com.saggitt.omega.OmegaPreferences.addOnPreferenceChangeListener(OmegaPreferences.kt:402)
	at com.saggitt.omega.OmegaPreferences.addOnPreferenceChangeListener(OmegaPreferences.kt:394)
	at com.saggitt.omega.views.SmartspacePreview.onAttachedToWindow(SmartspacePreview.kt:51)
	at android.view.View.dispatchAttachedToWindow(View.java:19803)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3448)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3455)
	at android.view.ViewGroup.addViewInner(ViewGroup.java:5251)
	at android.view.ViewGroup.addView(ViewGroup.java:5029)
	at android.view.ViewGroup.addView(ViewGroup.java:4969)
	at androidx.recyclerview.widget.RecyclerView$5.addView(RecyclerView.java:856)
	at androidx.recyclerview.widget.ChildHelper.addView(ChildHelper.java:107)
	at androidx.recyclerview.widget.RecyclerView$LayoutManager.addViewInt(RecyclerView.java:8601)
	at androidx.recyclerview.widget.RecyclerView$LayoutManager.addView(RecyclerView.java:8559)
	at androidx.recyclerview.widget.RecyclerView$LayoutManager.addView(RecyclerView.java:8547)
	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1641)
	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1587)
	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:665)
	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4134)
	at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:3851)
	at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4404)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1845)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at com.saggitt.omega.settings.DecorLayout$ContentFrameLayout.onLayout(DecorLayout.kt:211)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1845)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1845)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at com.android.internal.policy.DecorView.onLayout(DecorView.java:832)
	at android.view.View.layout(View.java:22190)
	at android.view.ViewGroup.layout(ViewGroup.java:6372)
	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3430)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2898)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2005)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8202)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1094)
	at android.view.Choreographer.doCallbacks(Choreographer.java:917)
	at android.view.Choreographer.doFrame(Choreographer.java:844)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1079)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:238)
	at android.app.ActivityThread.main(ActivityThread.java:7798)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:995)
Caused by: java.lang.IllegalStateException: Cannot call this method while RecyclerView is computing a layout or scrolling com.saggitt.omega.views.PreferenceRecyclerView{ca347bb VFED.V... ......ID 0,0-1080,2115 #7f0b016d app:id/list}, adapter:com.saggitt.omega.settings.HighlightablePreferenceGroupAdapter@76eaf67, layout:androidx.recyclerview.widget.LinearLayoutManager@43c4114, context:com.saggitt.omega.settings.SettingsActivity@a8e85eb
	at androidx.recyclerview.widget.RecyclerView.assertNotInLayoutOrScroll(RecyclerView.java:3051)
	at androidx.recyclerview.widget.RecyclerView$RecyclerViewDataObserver.onItemRangeChanged(RecyclerView.java:5547)
	at androidx.recyclerview.widget.RecyclerView$AdapterDataObservable.notifyItemRangeChanged(RecyclerView.java:12268)
	at androidx.recyclerview.widget.RecyclerView$Adapter.notifyItemChanged(RecyclerView.java:7398)
	at androidx.preference.PreferenceGroupAdapter.onPreferenceChange(PreferenceGroupAdapter.java:351)
	at androidx.preference.Preference.notifyChanged(Preference.java:1275)
	at androidx.preference.Preference.setSummary(Preference.java:754)
	at com.saggitt.omega.preferences.SmartspaceEventProvidersPreference.updateSummary(SmartspaceEventProvidersPreference.kt:54)
	at com.saggitt.omega.preferences.SmartspaceEventProvidersPreference.access$updateSummary(SmartspaceEventProvidersPreference.kt:34)
	at com.saggitt.omega.preferences.SmartspaceEventProvidersPreference$onListPrefChanged$1.invoke(SmartspaceEventProvidersPreference.kt:80)
	at com.saggitt.omega.preferences.SmartspaceEventProvidersPreference$onListPrefChanged$1.invoke(SmartspaceEventProvidersPreference.kt:34)
	at com.saggitt.omega.util.OmegaUtilsKt.runOnThread(OmegaUtils.kt:147)
	at com.saggitt.omega.util.OmegaUtilsKt.runOnMainThread(OmegaUtils.kt:142)
	at com.saggitt.omega.preferences.SmartspaceEventProvidersPreference.onListPrefChanged(SmartspaceEventProvidersPreference.kt:79)
	at com.saggitt.omega.OmegaPreferences$MutableListPref.saveChanges(OmegaPreferences.kt:564)
	at com.saggitt.omega.OmegaPreferences$MutableListPref.setAll(OmegaPreferences.kt:517)
	at com.saggitt.omega.smartspace.OmegaSmartspaceController.onProviderChanged(OmegaSmartspaceController.kt:193)
	at com.saggitt.omega.smartspace.OmegaSmartspaceController.<init>(OmegaSmartspaceController.kt:73)
	at com.saggitt.omega.OmegaApp$smartspace$2.invoke(OmegaApp.kt:40)
	at com.saggitt.omega.OmegaApp$smartspace$2.invoke(OmegaApp.kt:38)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.saggitt.omega.OmegaApp.getSmartspace(Unknown Source:2)
	at com.google.android.apps.nexuslauncher.smartspace.SmartspaceView.<init>(SmartspaceView.java:113)
	... 101 more

```

**Comment by aaronhatesregex** [comment]
Reproducible in v0.7.0 (build 115) on Android 8.0.0.

**Comment by otakuhqz** [comment]
Fixed in commit 7be0f0d
